src/Symfony/Component/Translation/Command/TranslationPushCommand.php

```diff
@@ -136,6 +136,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         if (!$domains) {
             if ($provider instanceof FilteringProvider) {
                 $domains = $provider->getDomains();
+                $localTranslations = $this->readLocalTranslations($locales, $domains, $this->transPaths);
```

Comment:
I guess it's better to not refresh translations, but read them after the `if (!$domains) {` block

Reply:
This was my first idea, but the variable `$localTranslations` is used in the ` if (!$domains) {` block at line 143 https://github.com/symfony/symfony/blob/c9d787d5a3bfe4a6c2f998ab3bc6d18e03073d31/src/Symfony/Component/Translation/Command/TranslationPushCommand.php#L143

We can also rewrite the` if (!$domains) {` block a bit like this :

```php
if (!$domains && $provider instanceof FilteringProvider) {
	$domains = $provider->getDomains();
}

$localTranslations = $this->readLocalTranslations($locales, $domains, $this->transPaths);

if (!$domains) {
	$domains = $this->getDomainsFromTranslatorBag($localTranslations);
}
```

Reply:
Yes, please

Reply:
Done